zerver/lib/user_topics.py

```diff
@@ -167,6 +167,40 @@ def set_user_topic_visibility_policy_in_database(
     return True
 
 
+@transaction.atomic(savepoint=False)
+def bulk_set_user_topic_visibility_policy_in_database(
+    user_profiles: List[UserProfile],
+    stream_id: int,
+    topic_name: str,
+    *,
+    visibility_policy: int,
+    recipient_id: Optional[int] = None,
+    last_updated: Optional[datetime.datetime] = None,
+) -> None:
+    if visibility_policy == UserTopic.VisibilityPolicy.INHERIT:
+        UserTopic.objects.filter(
+            user_profile__in=user_profiles,
+            stream_id=stream_id,
+            topic_name__iexact=topic_name,
+        ).delete()
+        return
+
+    assert last_updated is not None
+    assert recipient_id is not None
+
+    UserTopic.objects.bulk_create(
```

Comment:
As [mentioned](https://github.com/zulip/zulip/blob/main/zerver/actions/message_edit.py#L761)
```
    # We may want more complex behavior in cases where one appears to
    # be merging topics (E.g. there are existing messages in the
    # target topic).
```

So the current code assumes that there is no such case where the target topic has any messages (or **any UserTopic row**)

How should we handle the case when both original and target topics have some visibility_policy set?
* `ignore_conflicts=True` 
    * target-topic visibility_policy prevails?
    * prevails because that topic has discussions based on which user has configured visibility_policy. Messages moved there are of similar importance, hence similar visibility_policy
    * Now, if the user wants to change the visibility_policy, he/she can.
* `update_conflicts=True` (original_topic visibility_policy prevails; doesn't seem practical to me)

Currently it raises an error on manual testing.

Reply:
The third option is that we remove the `UserTopic` row (i.e set `visibility_policy=INHERIT`) when there is merging, so that the user can come to know about this (when the topic comes up somewhere in the UI) and decide what to do.

@alya I guess the above is more of a product/UX question. Could you take a look?

Reply:
[CZO discussion](https://chat.zulip.org/#narrow/stream/3-backend/topic/Unmute.20Topics/near/1533009)

I am planning to do this in a separate PR.
As we are extending the move-topic codepath to handle the discussed complex case and it doesn't affects this PR.
So, it would be better to do this separately? thoughts?

Reply:
I think a follow-up is fine for addressing the complexity, since this PR is not a regression to the behavior there.